mindsdb_native/libs/controllers/predictor.py

```diff
@@ -23,9 +23,9 @@ def _get_memory_optimizations(df):
 
     mem_usage_ratio = df_memory/total_memory
 
-    sample_for_analysis = True if mem_usage_ratio >= 0.05 else False
-    sample_for_training = True if mem_usage_ratio >= 0.15 else False
-    disable_lightwood_transform_cache = True if mem_usage_ratio >= 0.15 else False
+    sample_for_analysis = True if mem_usage_ratio >= 0.3 else False
```

Comment:
I agree that this should have been higher, but I think that, as per issue, this should actually be a function of the nr of rows up to a point (i.e. if we sample out of e.g. 500 rows, sampling might cause issues there). I will see how easy it is to get that logic in, do not merge for now, will merge when that is done.

Reply:
What if you have under 10k rows, but with 10k columns in each row?